README.md

```diff
@@ -28,6 +30,31 @@ JSON ESLint config example:
 }
 ```
 
+### Flat Configuration (`eslint-config.js`)
+
+Import the `eslint-plugin-github`, and extend any of the configurations using `getFlatConfigs()` as needed like so:
+
+```js
+import github from 'eslint-plugin-github'
+
+export default [
+  github.getFlatConfigs().browser,
+  github.getFlatConfigs().recommended,
+  github.getFlatConfigs().react,
+  ...github.getFlatConfigs().typescript,
```

Comment:
why is this one different than the reset?

Reply:
This was due to the way their recommended `tseslint.config` works as a part of [their upgraded `typescript-eslint`](https://typescript-eslint.io/blog/announcing-typescript-eslint-v7/#switching-to-typescript-eslint):

https://github.com/github/eslint-plugin-github/pull/571/files#diff-a9abd1556c7d3b1217538365a174c9d48cf9a18420eb7d92240409904d042ff0R6

It outputs an array while the others outputs objects (example, I shortened the number of rules for ease of reading):

Internal config:

```
{
  plugins: { github: { meta: [Object], rules: [Object] } },
  rules: {
    'github/authenticity-token': 'error',
    'github/js-class-name': 'error',
    'github/no-d-none': 'error'
  }
}
```

Typescript config:

```
[
  {
    rules: {
      'constructor-super': 'error',
      'for-direction': 'error',
    }
  },
  {
    name: 'typescript-eslint/base',
    languageOptions: { parser: [Object], sourceType: 'module' },
    plugins: { '@typescript-eslint': [Object] }
  },
  {
    files: [ '**/*.ts', '**/*.tsx', '**/*.mts', '**/*.cts' ],
    rules: {
      'constructor-super': 'off',
      'getter-return': 'off',
    },
    name: 'typescript-eslint/eslint-recommended'
  },
  {
    name: 'typescript-eslint/recommended',
    rules: {
      '@typescript-eslint/ban-ts-comment': 'error',
      'no-array-constructor': 'off',
    }
  },
  {
    languageOptions: { parser: [Object] },
    plugins: {
      '@typescript-eslint': [Object],
      escompatPlugin: [Object],
      github: [Object]
    },
    rules: {
      camelcase: 'off',
      'no-unused-vars': 'off',
    }
  }
]
```
So I use the spread operator to expand the elements in the typescript config array.

Reply:
ahh makes sense, thank you